rwengine/src/render/TextRenderer.cpp

```diff
@@ -230,24 +294,20 @@ void TextRenderer::renderText(const TextRenderer::TextInfo& ti,
             }
         }
 
-        auto& data = glyphData[glyph];
-        auto tex = indexToCoord(ti.font, glyph);
-
-        ss.x = ti.size * data.widthFrac;
-        tex.z = tex.x + (tex.z - tex.x) * data.widthFrac;
-
-        // Handle special chars.
-        if (c == '\n') {
-            coord.x = 0.f;
-            coord.y += ss.y;
-            maxHeight = coord.y + ss.y;
-            lineLength = 0;
-            continue;
-        }
+        auto tex = indexToTexCoord(glyph, fontMetaData.textureSize, fontMetaData.glyphOffset);
+        ss.x = ti.size * float(fontMetaData.glyphOffset.x) / fontMetaData.glyphOffset.y;
         lineLength++;
 
         glm::vec2 p = coord;
-        coord.x += ss.x;
+        float factor = ti.size / float(fontMetaData.glyphOffset.y);
```

Comment:
(here and below) It's not old style cast, but as bad as that in terms of maintainability.